doc/specs/stdlib_hashmaps.md

```diff
@@ -0,0 +1,2088 @@
+        procedure :: remove => remove_open_entry
+        procedure :: set_other_data => set_other_open_data
+        procedure :: total_depth => total_open_depth
+        final     :: free_open_map
+    end type open_hashmap_type
+```
+
+### Table of `stdlib_hashmap` procedures
+
+The `stdlib_hashmap` module provides procedures in
+several categories: a procedure to initialize the map; a procedure to
+modify the structure of a map; procedures to modify the content of a
+map; procedures to report on the content of a map; and procedures
+to report on the structure of the map. The procedures in each category
+are listed below.
+
+Procedure to initialize a chaining hash map:
+
+* `map % init( hasher[, slots_bits, status] )` - Routine
+  to initialize a chaining hash map.
+
+Procedure to modify the structure of a map:
+
+* `map % rehash( hasher )` - Routine to change the hash function
+  for a map.
+
+Procedures to modify the content of a map:
+
+* `map % map_entry( key, other, conflict )` - Inserts an entry into the
+  hash map.
+
+* `map % remove( key, existed )` - Remove the entry, if any,
+  associated with the `key`.
+
+* `map % set_other_data( key, other, exists )` - Change the other data
+  associated with the entry.
+
+Procedures to report the content of a map:
+
+* `map % get_other_data( key, other, exists )` - Returns the other data
+  associated with the `key`;
+
+* `map % key_test( key, present)` - Returns a flag indicating whether
+  the `key` is present in the map.
+
+Procedures to report on the structure of the map:
+
+* `map % calls()` - the number of subroutine calls on the hash map.
+
+* `map % entries()`- the number of entries in a hash map.
+
+* `map % loading()` - the number of entries relative to the number of
+  slots in a hash map.
+
+* `map % map_probes()` - the total number of table probes on a hash
+  map.
+
+* `map % slots()` - Returns the number of allocated slots in a hash
+  map.
+
+* `map % total_depth()` - Returns the total number of one's based
+offsets of slot entries from their slot index
+
+
+### Specifications of the `stdlib_hashmaps` procedures
+
+#### `calls` - Returns the number of calls on the hash map
+
+##### Status
+
+Experimental
+
+##### Description
+
+Returns the number of procedure calls on a hash map.
+
+##### Syntax
+
+`value = [[stdlib_hashmaps:map % calls]]()`
+
+##### Class
+
+Pure function
+
+##### Argument
+
+`map` (pass) - shall be an expression of class `hashmap_type`.
+It is an `intent(in)` argument.
+
+##### Result character
+
+The result will be an integer of kind `int_calls`.
+
+##### Result value
+
+The result will be the number of procedure calls on the hash map.
+
+##### Example
+
+```fortran
+    program demo_calls
+      use stdlib_hashmaps, only: chaining_hashmap_type, int_calls
+      use stdlib_hashmap_wrappers, only: fnv_1_hasher
+      implicit none
+      type(chaining_hashmap_type) :: map
+      type(int_calls) :: initial_calls
+      call map % init(fnv_1_hasher )
```

Comment:
```suggestion
      call map % init( fnv_1_hasher )
```
Just whitespace that feels odd, maybe it's time to discuss a CI enforced auto-style like  [fprettify](https://github.com/pseewald/fprettify) #3 and ways to avoid erroneous docs examples #639, so we can stop worrying about them.